Automattic/wp-calypso
# Design Picker: Refactor themes request to react-query and only fetch free themes

I've been meaning to refactor the theme requesting code out of the design picker step and into the `@automattic/design-picker` package. This is so code that's meant to convert the API format into the Design format can be in the package, and so the code can eventually be shared with Gutenboarding.

The premium themes project is a good excuse to do this refactoring because we needed to make changes to the Redux store to support making requests for different theme tiers. So if we were going to have to make changes to the data layer I figured we may as well make the change that'd allow the code to be shared.

We don't necessarily want premium themes to appear in the design picker as soon as they're available. Since users will have only just gone through the checkout we don't necessarily want to be upselling users again in the very next step, so we need to be more selective about whether to request premium themes or not. That's what #59293 is doing. Instead, this PR explicitly asks for only free themes, that way the premium themes project can ship whenever it's ready without us breaking.

#### Changes proposed in this Pull Request

* Move `<MShotsImage>` out of `design-picker` package and into `onboarding`. The ETK uses this component, but since the `design-picker` package now depends on the `lib/wp` networking library, I didn't want the ETK to have to pull that in just to use the mshots component.
* Add `useThemeDesignsQuery()` hook which follow the `react-query` pattern for requesting data. Called it "Theme Designs" because I figured it's possible that there could be other hooks (e.g. `useXyzDesignsQuery()`) that get things from other sources that also conform to the "Design" format.
* Design step requests themes using new hook
* Design step explicitly requests only free themes
* Set `is_premium` flag using network response
* Fixed hard coded DIFM themes now that they're expected to be in the Design format.

#### Testing instructions



* Test Hero Flow
  * `/start` and create a new free site
  * Test build or write flow
  * Navigate to the design picker step, scroll through themes, and confirm the network request for themes includes the `?tier=free` query param
  * Optionally you can opt-in to premium theme beta using the steps in D71467-code to confirm that no premium themes are shown in the design picker
* Test DIFM flow
  * `/start/do-it-for-me`
  * Ensure the DIFM design picker loads as before
  * Check network tab to ensure no request to the themes API is made
* Test Gutenboarding
  * Nothing's changed here. It still doesn't request designs from the theme API, but just smoke testing since there's been changes to the design picker package.
* Test change to ETK dependencies
  * Use `install-plugin.sh editing-toolkit update/fetch-design-picker-with-react-query` to install on sandbox
  * Create a new page on a sandboxed site
  * The page layout previews, and hover effects, should work just as before
